content/articles/django-celery-tasks/index.md

```diff
@@ -0,0 +1,333 @@
+
+@shared_task
+def name_of_your_function(optional_param):
+    pass  # do some long running task
+```
+
+#### Creating the HTML files
+
+Create a folder named `templates` in the projects root directory. In the `templates` directory created above, create another directory within it named `task` as it will hold the HTML files for our Django `task`.
+
+1. In the `task` directory created above, create a file named `base.html` and add the code snippets below into it. The below code snippet is the base template that other template files will extend from.
+   
+   ```html
+   <!DOCTYPE html>
+   <html>
+   <head>
+      <meta charset="utf-8">
+      <title>Celery tasks</title>
+      <style type="text/css">
+         body {
+         width: 800px;
+         margin: 20px auto;
+         }
+      </style>
+   </head>
+   <body>
+      <h1>Django Celery Task</h1>
+      <a href="{% url 'users_list' %}">Users List</a> /
+      <a href="{% url 'generate' %}">Generate Random Users</a>
+      <hr>
+
+      {% if messages %}
+         {% for message in messages %}
+            <p style="color: green">{{ message }}</p>
+         {% endfor %}
+      <hr>
+      {% endif %}
+
+   {% block content %}
+   {% endblock %}
+
+   </body>
+   </html>
+   ```
+   
+2. In the `task` directory, create a file named `user_list.html` and add the code snippets below. The below code snippet will display the list of generated random users.
+   
+   ```html
+   {% extends 'task/base.html' %}
+
+   {% block content %}
+    <h2>Users List</h2>
+
+    <ul>
+        {% for user in object_list %}
+            <li>{{ user.username }} - {{ user.email }} - {{ user.date_joined }}</li>
+        {% empty %}
+            <li>No users. <a href="{% url 'generate' %}">Generate some random users.</a></li>
+        {% endfor %}
+    </ul>
+   {% endblock %}
+   ```
+  
+3. In the `task` directory create a file named `generate_random_user.html` and add the code snippets below. The below code snippet contains an input field where we will specify the number of random users to generate.
+   
+   ```html
+   {% extends 'task/base.html' %}
+
+   {% block content %}
+    <h2>Generate Random Users</h2>
+    <form method="post">
+        {% csrf_token %}
+        <table>
+            {{ form }}
+        </table>
+        <button type="submit">Submit</button>
+    </form>
+   {% endblock %}
+   ```
+
+#### Form
+
+In the `task`, create a python file named `random_user_form.py` and add the code snippet below:
+
+```python
+from django import forms
+from django.core.validators import MinValueValidator, MaxValueValidator
+
+class GenerateRandomUserForm(forms.Form):
+    total = forms.IntegerField(
+        validators=[
+            MinValueValidator(50),
+            MaxValueValidator(500)
+        ]
+    )
+```
+
+### View
+
+Add the code snippet below into the `views.py` file in the `task`:
+
+```python
+from django.shortcuts import redirect
+
+from .form import GenerateRandomUserForm
```

Comment:
This import statement is wrong. Either rename the file or fix this import statement,